prov/efa/src/efa_cq.h

```diff
@@ -94,23 +94,23 @@ void efa_ibv_cq_poll_list_remove(struct dlist_entry *poll_list, struct ofi_genlo
  *
  * @param[in] ibv_cq_init_attr_ex Pointer to ibv_cq_init_attr_ex
  * @param[in] ibv_ctx Pointer to ibv_context
- * @param[in,out] ibv_cq_ex Pointer to newly created ibv_cq_ex
- * @param[in,out] ibv_cq_ex_type enum indicating if efadv_create_cq or ibv_create_cq_ex was used
+ * @param[in,out] ibv_cq Pointer to newly created efa_ibv_cq
  * @return Return 0 on success, error code otherwise
  */
 static inline int efa_cq_ibv_cq_ex_open_with_ibv_create_cq_ex(
 	struct ibv_cq_init_attr_ex *ibv_cq_init_attr_ex,
-	struct ibv_context *ibv_ctx, struct ibv_cq_ex **ibv_cq_ex,
-	enum ibv_cq_ex_type *ibv_cq_ex_type)
+	struct ibv_context *ibv_ctx, struct efa_ibv_cq *ibv_cq)
 {
-	*ibv_cq_ex = ibv_create_cq_ex(ibv_ctx, ibv_cq_init_attr_ex);
+	*ibv_cq = (struct efa_ibv_cq) {
+		.ibv_cq_ex = ibv_create_cq_ex(ibv_ctx, ibv_cq_init_attr_ex),
+		.ibv_cq_ex_type = IBV_CQ,
+	};
 
-	if (!*ibv_cq_ex) {
+	if (!ibv_cq->ibv_cq_ex) {
 		EFA_WARN(FI_LOG_CQ, "Unable to create extended CQ: %s\n", strerror(errno));
 		return -FI_EINVAL;
 	}
 
-	*ibv_cq_ex_type = IBV_CQ;
```

Comment:
This is a behavior change, that makes `ibv_cq_ex_type` always set as `IBV_CQ`. Are you sure it will not impact the fall back path (non-extended cq)?

Reply:
Yeah this shouldn't change anything, but the whole commit can be dropped regardless